tests/lib/thrift/ThriftTest/src/client.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2022 Young Mei
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+
+#include <array>
+#include <cfloat>
+#include <locale>
+#include <stdexcept>
+
+#include <sys/time.h>
+
+#include "context.hpp"
+
+using namespace apache::thrift;
+
+using namespace std;
+
+static void init_Xtruct(Xtruct& s);
+
+ZTEST(thrift, test_void) {
+  context.client->testVoid();
+}
+
+ZTEST(thrift, test_string) {
+  string s;
+  context.client->testString(s, "Test");
+  zassert_equal(s, "Test", "");
+}
+
+ZTEST(thrift, test_bool) {
+  zassert_equal(false, context.client->testBool(false), "");
+  zassert_equal(true, context.client->testBool(true), "");
+}
+
+ZTEST(thrift, test_byte) {
+  zassert_equal(0, context.client->testByte(0), "");
```

Comment:
Formatting of files is not correct, should be using tabs not spaces

Reply:
> Formatting of files is not correct, should be using tabs not spaces

Assuming that `zephyr` has formatting rules for C++ now, I'll add another commit that enforces them.

`gsoc-2022-thrift` actually added C++ formatting rules prior to `zephyr` having them 😄

Reply:
There is an issue open for the C++ coding (formatting) guidelines: https://github.com/zephyrproject-rtos/zephyr/issues/45195.

Although we have not reached a conclusion, 2 spaces for indentation is quite common for C++ source files and is a likely good candidate for future adoption.

These C++ source files can be re-formatted in the future once the C++ coding guidelines are finalised and in effect.